readme.md

```diff
@@ -3,71 +3,704 @@
+
+Here we will need to define the folders and files that we want bundled, as well as any additional functionality from our tool. We are going to add in a few pieces of code:
+
+```js
+const path = require('path');
+
+const PATHS = {
+  js: path.join(__dirname, 'js'),
+  build: path.join(__dirname, 'build')
+};
+```
+
+<details>
+<summary>
+What do you think this first part is doing?
+</summary>
+<br>
+
+```
+This simply defines two folders within our app that we will be either reading or modifying with webpack.
+
+*Note* 'path' is a node method
+```
+<br>
+<br>
+</details>
+
+```js
+module.exports = {
+  // Entry accepts a path or an object of entries. We'll be using the
+  // latter form given it's convenient with more complex configurations.
+  entry: {
+    js: PATHS.js
+  }
+```
+
+<details>
+<summary>
+What do you think this next section does?
+</summary>
+<br>
+
+```
+Here we are defining the entry point of our webpack. In other words, whatdirectory do we want to look into and bundle?
+```
+<br>
+<br>
+</details>
+
+Lastly, we will need to define the output. Where are we going to put thebundled code?
+
+```js
+,
+  output: {
+    path: PATHS.build,
+    filename: 'bundle.js'
+  }
+};
+```  
+
+**Try running `webpack` in your terminal again. What happens?**
+
+Check out your public folder and see what file was added in!
+
+## STOP
+
+### Setting up Webpack Server
+
+Having to run `webpack` every time you make a change will get frustrating (and boring) quickly. We can set up the `webpack-dev-server` to help us out!
+
+One of the best features of the dev server is Hot Module Replacement (HMR). The module will allow us to make changes, and see them automatically without a full page refresh.
 
+To get started, run the following in you command line:
+
+`npm i webpack-dev-server --save-dev`
+
+We can now add this directly to our `package.json` file, allowing us the ability to start up our application immediately on the webpack dev server. Include the following in your `scripts` object in the `package.json` file:
+
+`"start": "webpack-dev-server --content-base build"`
+
+Go ahead and run `npm start` in your terminal. You should see something like:
+
+![npm Start](npmStart.png)
+
+If you open your browser to `localhost:8080`, you should see your index.html and any html rendered through your index.js file on the browser!
+
+This is cool and all, but not really any different than before other than running a different command in your terminal. We want to be able to make changes to our code and see the updates without having to do any additional refreshing.
+
+We will have to install another package to get this up and running: `npm i webpack-merge --save-dev`
+
+This package allows us to merge objects together. In this case, we will be merging the object that sets our paths, with HMR.
+
+In webpack.config.js, include the following on line 2:
+
+```js
+const merge = require('webpack-merge');
```

Comment:
apparently it allows for more optimized compiling. I saw the code written with both var and const; figured it might be something interesting to discuss for a minute. When we figure out timing, i might change it back to var!